docker/ubuntu-14.04-cpu-all-options/Dockerfile

```diff
@@ -0,0 +1,72 @@
+FROM ubuntu:14.04
+MAINTAINER Aaron Markham <aaronmarkham@fb.com>
+
+# caffe2 install with cpu support
+
+########## REQUIRED DEPENDENCIES ################
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    python-dev \
+    python-pip \
+    git \
+    build-essential \
+    cmake \
+    libprotobuf-dev \
+    protobuf-compiler \
+    libgoogle-glog-dev
+
+RUN git clone --recursive https://github.com/caffe2/caffe2.git && \
```

Comment:
Consider moving the code clone to the end of the script. The code is likely to change faster than the package dependencies, and you'll lose your docker build cache.